File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> answer = new ArrayList<>();
        int newNum = -1;
        
        for (int el : arr) {
            if (newNum != el) {
                newNum = el;
                answer.add(el);    
            }
        }

        return answer.stream()
                    .mapToInt(i -> i)
                    .toArray();
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    
    i = 1
    j = 0
    new = []
    new.append(arr[0])
    
    while i < len(arr):
        if new[j] != arr[i]:
            new.append(arr[i])
            j += 1
        i += 1
        
    
    return new